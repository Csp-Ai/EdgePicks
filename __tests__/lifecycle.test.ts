import fs from 'fs';
import path from 'path';

describe('agent lifecycle logs', () => {
  it('matches snapshot', () => {
    const file = path.join(__dirname, '..', 'agentLogsStore.json');
    const logs = JSON.parse(fs.readFileSync(file, 'utf8'));
    expect(logs).toMatchSnapshot();
  });
});
