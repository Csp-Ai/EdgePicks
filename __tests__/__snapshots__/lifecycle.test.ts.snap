// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`agent lifecycle logs matches snapshot 1`] = `{}`;
=======
exports[`agent lifecycle logs matches snapshot 1`] = `
[
  {
    "architectureDocumented": true,
    "codexRegistered": true,
    "command": "npx ts-node --compiler-options '{"module":"CommonJS"}' scripts/docsync-agent.ts",
    "designPatternsDocumented": true,
    "lifecycleDocumented": true,
    "output": "Error: supabaseUrl is required.",
    "reflectionLogged": true,
    "syncAttemptedAt": "2025-08-07T10:25:55.847Z",
    "syncError": "supabase env missing",
    "synced": false,
    "timestamp": "2025-08-07T10:17:20Z",
  },
  {

    "command": "npx ts-node scripts/uiSnapshot.ts",
    "snapshotTaken": true,
    "timestamp": "2025-08-07T12:00:00Z",
  },
  {
    "architectureDocumented": true,
    "codexRegistered": true,
    "command": "npx ts-node scripts/docsync-agent.ts",
    "designPatternsDocumented": true,
    "lifecycleDocumented": true,
    "output": "GH_PAT is required.",
    "syncAttemptedAt": "2025-08-07T11:05:00Z",
    "syncError": "GH_PAT is required.",
    "synced": false,
    "timestamp": "2025-08-07T11:05:00Z",
=======
    "architectureDocumented": true,
    "codexRegistered": true,
    "command": "npx ts-node scripts/uiSnapshot.ts",
    "snapshotTaken": true,
    "timestamp": "2025-08-07T12:00:00Z",

  },
  {
    "codexRegistered": true,
    "command": "npx ts-node --compiler-options '{"module":"CommonJS"}' scripts/audit-agent-metadata.ts",
    "metadataAudited": true,
    "output": "Agent metadata is valid and docs are in sync.",
    "timestamp": "2025-08-07T10:58:40Z",
  },
]
`;
